### DISCUSSION TITLE
Map reload required when coming back to map in chrome on android

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre-gl-js#3629] [Q&A]
I wanted to ask this question for a while or wanted to even file a bug report but I never figured out a good report so let me just try to describe the problem:

* First say I load a MapLibre GL JS map in Chrome on Android. All is fine and I pan around.
* Then say I don't use my phone for 1 hour. And in the mean time I have lost mobile connection so now I am offline.
* Now sometimes the map still works and I can zoom and pan around with the tiles I have in cache.
* But sometimes also the map just triggers a reload. In that case, because I am offline, my map is gone.

So my question is: have others observed the same behavior? And if yes, is that a MapLibre-related problem or is that something that Chrome does?

I never have this problem with other websites. For example news articles I am always able to read even when the phone goes offline...

**Comment by HarelM** [comment]
There are a few aspects here:
1. Cache control of the tiles
2. Browser behavior.

For the second one there's not much you can do, but I don't know if the browser actually refreshes the page.

For the first one there are things that you can do, but offline support is a wide subject, service workers are generally your fried here and other technics.
Not sure there's an out of the box solution as you need to define which behavior you want to achieve.

**Reply by wipfli** [reply]
Maybe I can see how Leaflet compares on openstreetmap.org and then see if that behavior is more of what I am looking for...

**Reply by HarelM** [reply]
Note that leaflet is using img and svg html elements, but yes, the UX is what we are after here.
